File: dotnet/WebService/src/WebApi/Controllers/TestsController.cs
using System;
using System.Net.Mime;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using WebApi.Dtos;
using WebApi.Utils;

namespace WebApi.Controllers
{
    [Route("api")]
    [ApiController]
    public class TestsController : ControllerBase
    {
        [HttpGet("json")]
        [Produces(MediaTypeNames.Application.Json)]
        [ProducesResponseType(StatusCodes.Status200OK)]
        public ActionResult<JsonTestDto> GetJson()
        {
            var jsonTestDto = new JsonTestDto
            {
                Message = $"API is working! Path: {HttpContext.Request.Path}",
                Date = DateTime.Now.ToString("yyyy-MM-ddTHH:mm:ss.ffffffZ")
            };

            return Ok(jsonTestDto);
        }

        [HttpGet("plaintext")]
        [Produces(MediaTypeNames.Text.Plain)]
        [ProducesResponseType(StatusCodes.Status200OK)]
        public ActionResult<string> GetPlainText()
        {
            string message = $"API is working! Path: {HttpContext.Request.Path}";

            return Ok(message);
        }

        [HttpGet("base64")]
        [Produces(MediaTypeNames.Application.Json)]
        [ProducesResponseType(StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status400BadRequest)]
        public ActionResult<Base64Dto> GetBase64([FromQuery] string message = "This is default message.")
        {
            int requiredMinLength = 3;
            int requiredMaxLenth = 25;

            if (message == null || message.Length < requiredMinLength || message.Length > requiredMaxLenth)
            {
                var errorMessageDto = new ErrorMessageDto
                {
                    Message = $"'message' length must be between {requiredMinLength} and {requiredMaxLenth}."
                };

                return BadRequest(errorMessageDto);
            }

            var base64Dto = new Base64Dto
            {
                Message = message,
                EncodedMessage = Base64.Encode(message)
            };
            base64Dto.DecodedMessage = Base64.Decode(base64Dto.EncodedMessage);

            return Ok(base64Dto);
        }
    }
}

File: express/webservice/src/controllers/testsController.js
import moment from 'moment';
import Base64Dto from '../dtos/base64Dto.js';
import ErrorMessageDto from '../dtos/errorMessageDto.js';

const json = (req, res) => {
    const jsonTestDto = {
        message: `API is working! Path: ${req.path}`,
        date: moment().format('YYYY-MM-DDTHH:mm:ss.SSS')
    };

    res.status(200).json(jsonTestDto);
};

const plaintext = (req, res) => {
    const message = `API is working! Path: ${req.path}`;

    res.status(200).set('Content-Type', 'text/plain').send(message);
};

const base64 = (req, res) => {
    const message = req.query.message || 'This is default message.';

    const requiredMinLength = 3;
    const requiredMaxLenth = 25;

    if (message === null || message.length < requiredMinLength || message.length > requiredMaxLenth) {
        const errorMessageDto = new ErrorMessageDto(`'message' length must be between ${requiredMinLength} and ${requiredMaxLenth}.`);
        
        return res.status(400).json(errorMessageDto);
    }

    const base64Dto = new Base64Dto(message);

    res.status(200).json(base64Dto);
};

const testsController = {
    json,
    plaintext,
    base64
};

export default testsController;
